sklearn/cluster/optics_.py

```diff
@@ -491,12 +492,21 @@ def _calculate_optics_order(self, X, nbrs):
         for point in range(X.shape[0]):
             if processed[point]:
                 continue
-            if self.core_distances_[point] <= self.max_eps:
-                while not processed[point]:
+            if self.core_distances_[point] != np.inf:
+                while True:
                     processed[point] = True
                     ordering[ordering_idx] = point
                     ordering_idx += 1
-                    point = self._set_reach_dist(point, processed, X, nbrs)
+                    if self.core_distances_[point] != np.inf:
+                        self._set_reach_dist(point, processed, X, nbrs)
+                    # Take all the points which have not been processed
+                    # and need to be processed in current while loop.
+                    mask = processed | (self.reachability_ == np.inf)
```

Comment:
`core_distance <= max_eps` is equivalent since it is `np.inf` exactly if it is `> max_eps`.
Maybe this needs to be a `while self.core_distances_[point] <= self.max_eps:` to check this conditions for *every* point.